moz/l10n/resource/parse_resource.py

```diff
@@ -0,0 +1,63 @@
+# Copyright Mozilla Foundation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from ..message import Message
+from .android.parse import android_parse
+from .data import Resource
+from .dtd.parse import dtd_parse
+from .fluent.parse import fluent_parse
+from .format import Format, detect_format
+from .inc.parse import inc_parse
+from .ini.parse import ini_parse
+from .plain_json.parse import plain_json_parse
+from .po.parse import po_parse
+from .properties.parse import properties_parse
+from .webext.parse import webext_parse
+from .xliff.parse import xliff_parse
+
+
+def parse_resource(
+    type: Format | str | None, source: str | bytes
```

Comment:
Why is the `type` parameter required?

I would expect the file path to be often passed to the function implicitly through the `source` parameter. It seems redundant to then be required to also include `type=None` as the function parameter.

Reply:
Because `source` holds the actual contents of the file, it won't have the file path. So you can pass the path in with `type: str`, as in `parse_resource('/foo/bar.json', source)`.

Reply:
> So you can pass the path in with `type: str`, as in `parse_resource('/foo/bar.json', source)`.

And that's the thing that feels redundant - `parse_resource('/foo/bar.json')` should be sufficient.

Reply:
That makes sense. I made the second argument optional, if the first one is a string path.